### PR TITLE
docs(design): Write spacing guidelines

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations
We've never had any guidelines or examples of how to use Spacing. We should have that, so that designers can follow defined patterns and we're not just doing what seems good on a case-by-case basis.

### Added

- Adds usage guidelines to Spacing.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by lorborb** [comment]
Nice @chris-at-jobber! Could I maybe make a request (sorry) to add some sort of colour/signifier to the spacing you're referring to in the screenshots? Word-wise I think everything is pretty well described

**Comment by chris-at-jobber** [comment]
Thanks @lorborb , preview should be ready with the annotated example images in a few minutes 😃

**Comment by chriscarline** [comment]
Is there a way to more concretely describe the usage of `large`? It might just need to live in the abstract world of one thing needing to breathe more but firmer guidelines/other examples would be nice IMO. 
image

**Comment by chriscarline** [comment]
Also noticed some small typo/grammar things:

- `Between` is missing its `t` in the Implementation content section at the bottom
- I believe `its'` should be `its`
- I don't think you need that second comma in `shortest, simplest, terms`

File: packages/design/src/Spacing.mdx [hunk 1]
```diff
@@ -6,7 +6,76 @@ route: /space
...
 # Spacing
 
-Spacing for the Jobber Atlantis Design System
+Our spacing system exists to support the foundational visual design 
+principle of [proximity-based grouping:](https://www.nngroup.com/articles/gestalt-proximity/) 
+
+> Using varying amounts of whitespace to either unite or separate elements 
+> is key to communicating meaningful groupings.
+
+In the shortest, simplest, terms: the closer two elements are related,
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:15]:
```suggestion
In the shortest, simplest terms: the closer two elements are related, 
```

File: packages/design/src/Spacing.mdx [hunk 2]
```diff
@@ -6,7 +6,76 @@ route: /space
...
+You’ll probably never need these unless you’re doing a custom layout, but they’re here if you need ‘em.
+These values are equivalent to `48px` and `64px` respectively.
+
+---
+
+## Implementation
+
+### Content
+
+Use the [Content component](/components/content) for any instances where you need uniform spacing beween elements.
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:74]:
```suggestion
Use the [Content component](/components/content) for any instances where you need uniform spacing between elements.
```

File: packages/design/src/Spacing.mdx [hunk 3]
```diff
@@ -6,7 +6,76 @@ route: /space
...
+
+Content can set its' spacing to either `small`, `base`, or `large`; follow the guidelines above as needed.
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:76]:
```suggestion
Content can set its spacing to either `small`, `base`, or `large`; follow the guidelines above as needed.
```

File: packages/design/src/Spacing.mdx [hunk 4]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+### Base
+
+Use `base` as the default spacing in larger content containers with multiple
+children of their own, such as [Card.](/components/card) This should also be
+used as the default starting point for spacing between [Form](/components/form)
+elements.
+
+![illustration of spacing between form elements inside of a card](https://user-images.githubusercontent.com/39704901/143141662-0f20eec4-8b27-44fe-96f5-1bf1c4d567aa.png)
+
+### Large
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:53]: @chriscarline I _think_ this is more specific but still generalizable? Lemme know what you think.

Comment (chris-at-jobber) [reply packages/design/src/Spacing.mdx:53]: Preview: https://deploy-preview-779--jobber-atlantis.netlify.app/space#large

Comment (chriscarline) [reply packages/design/src/Spacing.mdx:53]: @chris-at-jobber makes sense to me! Just need to drop those two its'

File: packages/design/src/Spacing.mdx [hunk 5]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+
+Use `large` spacing for higher-level "parent" containers.
+[Modal](/components/modal) uses `large` spacing in its' header and around the
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:56]:
```suggestion
[Modal](/components/modal) uses `large` spacing in its header and around the
```

File: packages/design/src/Spacing.mdx [hunk 6]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+edges to give itself a bit more prominence as a "standalone" view, in a similar
+manner to [Page](/components/page).
+
+`Large` can also be used to help give a "floating" element some clear separation
+from the edge of its' container, as seen in [Toast](/components/toast).
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:61]:
```suggestion
from the edge of its container, as seen in [Toast](/components/toast).
```

File: packages/design/src/Spacing.mdx (hunk 4 above)
Comment (chris-at-jobber) [reply packages/design/src/Spacing.mdx:53]: ✅

**Comment by chriscarline** [comment]
Splendid!

**Review by chriscarline — APPROVED** [review]

**Comment by lorborb** [comment]
Since Carline was looking for grammar, now I am also looking for grammar. Only found, third paragraph (intro) "With in mind" add 'that'. All else is excellence.

File: packages/design/src/Spacing.mdx [hunk 7]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+Our spacing system exists to support the foundational visual design principle of
+[proximity-based grouping:](https://www.nngroup.com/articles/gestalt-proximity/)
+
+> Using varying amounts of whitespace to either unite or separate elements is
+> key to communicating meaningful groupings.
+
+In the shortest, simplest terms: the closer two elements are related, the
+tighter the spacing between them should be.
+
+With in mind, let’s start small and work our way up.
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:18]:
```suggestion
With this in mind, let’s start small and work our way up.
```

File: packages/design/src/Spacing.mdx [hunk 8]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+Our spacing system exists to support the foundational visual design principle of
+[proximity-based grouping:](https://www.nngroup.com/articles/gestalt-proximity/)
+
+> Using varying amounts of whitespace to either unite or separate elements is
+> key to communicating meaningful groupings.
+
+In the shortest, simplest terms: the closer two elements are related, the
+tighter the spacing between them should be.
+
+With this in mind, let’s start small and work our way up.
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:18]: @lorborb ✅

File: packages/design/src/Spacing.mdx [hunk 9]
```diff
@@ -6,7 +6,89 @@ route: /space
...
+
+## Design & usage guidelines
+
+### Miniscule and Smallest
+
+At their respective sizes (`1px` and `2px` equivalent), these values should
+mostly be used to provide optical adjustments where an element is visually
+off-centered.
```
Comment (chris-at-jobber) [inline packages/design/src/Spacing.mdx:26]:
```suggestion
misaligned.
```

**Pull request merged**